test/RibbonThetaVaultYearn.js

```diff
@@ -0,0 +1,3531 @@
+            .connect(userSigner)
+            .deposit({ value: parseEther("1") });
+          await this.assetContract
+            .connect(userSigner)
+            .transfer(this.vault.address, parseEther("1"));
+
+          await this.rollToNextOption();
+
+          // formula:
+          // (depositAmount * totalSupply) / total
+          // (1 * 3) / 4 = 0.75 shares
+          await this.vault
+            .connect(counterpartySigner)
+            .depositETH({ value: parseEther("1") });
+          assert.equal(
+            (await this.vault.balanceOf(counterparty)).toString(),
+            parseEther("0.75")
+          );
+        });
+
+        it("reverts when no value passed", async function () {
+          await expect(
+            this.vault.connect(userSigner).depositETH({ value: 0 })
+          ).to.be.revertedWith("No value passed");
+        });
+
+        it("does not inflate the share tokens on initialization", async function () {
+          await this.assetContract
+            .connect(adminSigner)
+            .deposit({ value: parseEther("10") });
+          await this.assetContract
+            .connect(adminSigner)
+            .transfer(this.vault.address, parseEther("10"));
+
+          await this.vault
+            .connect(userSigner)
+            .depositETH({ value: parseEther("1") });
+
+          // user needs to get back exactly 1 ether
+          // even though the total has been incremented
+          assert.isFalse((await this.vault.balanceOf(user)).isZero());
+        });
+
+        it("reverts when minimum shares are not minted", async function () {
+          await expect(
+            this.vault.connect(userSigner).depositETH({
+              value: BigNumber.from("10").pow("10").sub(BigNumber.from("1")),
+            })
+          ).to.be.revertedWith(/Insufficient asset balance/);
+        });
+      });
+    }
+
+    describe("#deposit", () => {
+      time.revertToSnapshotAfterEach();
+
+      beforeEach(async function () {
+        // Deposit only if asset is WETH
+        if (params.depositAsset === WETH_ADDRESS) {
+          const addressToDeposit = [
+            userSigner,
+            managerSigner,
+            counterpartySigner,
+            adminSigner,
+          ];
+
+          for (let i = 0; i < addressToDeposit.length; i++) {
+            const weth = this.assetContract.connect(addressToDeposit[i]);
+            await weth.deposit({ value: parseEther("10") });
+            await weth.approve(this.vault.address, parseEther("10"));
+          }
+        }
+      });
+
+      it("deposits successfully", async function () {
+        const depositAmount = BigNumber.from("100000000000");
+
+        await this.assetContract
+          .connect(userSigner)
+          .approve(this.vault.address, depositAmount);
+
+        const res = await this.vault.deposit(depositAmount);
+        const receipt = await res.wait();
+        assert.isAtMost(receipt.gasUsed.toNumber(), 150000);
+
```

Comment:
Check that `totalBalance` and `assetBalance` values are incremented correctly after a deposit